src/main/java/Duke.java

```diff
@@ -1,10 +1,138 @@
+            }
+        }
+        printSingleLine();
+    }
+
+    private static void printDetailsOfAddedTask(Task newTask) {
+        System.out.println("One more thing you got to do. Press on!");
+        System.out.println(newTask.toString());
+        System.out.println("Now you have " + Task.getNumberOfTask() + " tasks in the list.");
+        printSingleLine();
+    }
+
+    private static void updateTaskAsComplete(Task[] tasks, String description) {
+        int taskNumber = Integer.parseInt(description);
+        tasks[taskNumber].markAsDone();
+        printDetailsOfCompletedTask(tasks[taskNumber]);
+    }
+
+    private static void printDetailsOfCompletedTask(Task task) {
+        System.out.println("Nice. One more down!");
+        System.out.println(task.toString());
+        printSingleLine();
+    }
+
+    private static void updateListOfTask(String command, Task[] tasks) {
+        int indexToSplit = command.indexOf(' ');
+        if (indexToSplit == -1) {
+            printInvalidTaskMessage();
+        } else {
+            String typeOfTask = command.substring(0, indexToSplit);
+            String description = command.substring(indexToSplit);
+            description = description.trim();
+
+            switch (typeOfTask) {
+            case "todo":
+                createTodoTask(tasks, description);
+                break;
+            case "deadline":
+                createDeadlineTask(tasks, description);
+                break;
+            case "event":
+                createEventTask(tasks, description);
+                break;
+            case "done":
+                updateTaskAsComplete(tasks, description);
+                break;
+            default:
+                printInvalidTaskMessage();
+                break;
+            }
+        }
+    }
+
+    private static void createTodoTask(Task[] tasks, String description) {
+        Task newTask = new ToDo(description);
+        tasks[Task.getNumberOfTask()] = newTask;
+        printDetailsOfAddedTask(newTask);
+    }
+
+    private static void createDeadlineTask(Task[] tasks, String description) {
+        String[] deadlineDetails = description.split("/by");
+        String deadline = deadlineDetails[1].trim();
+        Task newTask = new Deadline(deadlineDetails[0], deadline);
+        tasks[Task.getNumberOfTask()] = newTask;
+        printDetailsOfAddedTask(newTask);
+    }
+
+    private static void createEventTask(Task[] tasks, String description) {
+        String[] eventDetails = description.split("/at");
+        String eventDateTime = eventDetails[1].trim();
+        Task newTask = new Event(eventDetails[0], eventDateTime);
+        tasks[Task.getNumberOfTask()] = newTask;
+        printDetailsOfAddedTask(newTask);
+    }
+
+    private static void printInvalidTaskMessage() {
+        System.out.println("I'm sorry I don't understand you." + System.lineSeparator()
+                + "Would you like to tell me again?");
+        printSingleLine();
```

Comment:
Great job on extracting your code into different readable functions with clear function heading.
And also a fantastic job on error handling.